site_tools/image_deployer.py

```diff
@@ -0,0 +1,297 @@
+    return response.text.strip()
+
+
+def get_available_images(
+        storage_url,
+        base_container,
+        scope,
+        current,
+        image_type):
+    available_images = []
+
+    current_path = f"{scope}/{current}"
+    url = f"{storage_url}/{base_container}?prefix={current_path}"
+    logging.info(f"Checking available images at {url}...")
+    response = requests.get(url)
+    if response.status_code != 200:
+        raise Exception(f"Error getting available images: {response.content}")
+
+    current_objects = response.text.splitlines()
+    logging.debug(f"Current objects: {current_objects}")
+    for object in islice(current_objects, 1, None):
+        object_name = object.split("/")[-1]
+        if object_name.endswith(".manifest"):
+            name = object_name.rstrip(".manifest")
+            available_images.append(
+                Image(name, image_type, base_container, scope, current_path)
+            )
+
+    return available_images
+
+
+def get_site_images(connection):
+    logging.info("Checking public site images...")
+    images = [
+        i.name
+        for i in connection.image.images(
+            visibility="public"
+        )
+    ]
+    return images
+
+
+def should_sync_image(image_disk_name, site_images, current):
+    if image_disk_name in site_images:
+        logging.info(f"Image {image_disk_name} already in site images.")
+        image = image_connection.image.find_image(image_disk_name)
+        image_properties = image.properties
+        logging.debug(f"Image properties: {image_properties}")
+        image_current_value = image_properties.get("current", None)
+        logging.debug(f"Image {image_disk_name} current value: {image_current_value}")
+        if image_current_value is not None and image_current_value == current:
+            logging.info(f"Image {image_disk_name} is already current.")
+            return False
+    return True
+
+
+def download_object_to_file(storage_url, path, file_name):
+    url = f"{storage_url}/{path}/{file_name}"
+    response = requests.get(url, stream=True)
+
+    if response.status_code != 200:
+        raise Exception(f"Error downloading object {file_name}: {response.content}")
+
+    with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+        for chunk in response.iter_content(chunk_size=65536):
+            temp_file.write(chunk)
+            temp_file.flush()
+    logging.debug(f"Downloaded object to {temp_file.name}.")
+    return temp_file
+
+
+def upload_image_to_glance(image_connection,
+                           image_prefix,
+                           image_disk_name,
+                           image_file_name,
+                           disk_format,
+                           manifest_data):
+    image_prefix_name = image_prefix + image_disk_name
+
+    # TODO: need admin creds to make visibility public instead of private
+    # right now I am using app creds that just have a member role
```

Comment:
since we're testing on uc_dev, just grab an app credential scoped to the `openstack` project, should be fine for testing this.